### DISCUSSION TITLE
Triangulate a 3D mesh

### DISCUSSION THREAD

**Discussion by boxerab** [libigl/libigl#2083] [Q&A]
I am using libigl to create a 3D mesh. Is it possible to triangulate the mesh i.e. triangulate each face ? Faces that are already
triangulated should be ignored. Any guidance or advice on this would be greatly appreciated !!

**Comment by alecjacobson** [comment]
Maybe you're looking for igl::polygons_to_triangles

**Reply by boxerab** [reply]
Thanks. It looks like this method adds a fan of triangles to the polygon - something that I could also do by hand.
> Given a polygon mesh, trivially triangulate each polygon with a fan
What I was hope for was to distribute the edges in a balanced way.

https://docs.blender.org/manual/en/latest/modeling/modifiers/generate/triangulate.html
